src/manifolds/EmbeddedManifold.jl

```diff
@@ -0,0 +1,88 @@
+"""
+    AbstractEmbeddedManifold
+
+"""
+abstract type AbstractEmbeddedManifold <: AbstractDecoratorManifold end
+
+abstract type AbstractEmbeddingType end
+abstract type AbstractIsometricEmbedding <: AbstractEmbeddingType end
+
+"""
+    IsometricEmbedding <: AbstractEmbeddingType
+
+"""
+struct DefaultIsometricEmbedding <: AbstractIsometricEmbedding end
+
+struct EmbeddedManifold{MT <: Manifold, NT <: Manifold, ET <: AbstractEmbeddingType} <: AbstractEmbeddedManifold
+    submanifold::MT
+    manifold::NT
```

Comment:
`EmbeddedManifold` is decorates `submanifold` in this case, right? Because in decorators we were assuming the the field `.manifold` is the decorated manifold. This may be confusing. Maybe we can call this `manifold` and `ambient_manifold`?

Reply:
Well, I first had the fields `manifold` and `embedding`. Then I noticed that most functions we have we want to `pass down` to the embedding, so to avoid reimplementation of all functions, it's best to see `EmbeddedManifold` and another view on the `embedding`, i.e. restricting it to a submanifold. The only thing that does not yet work is basis, vector and coordinates. Those might have to be implemented, still.

Reply:
Good point but it's somewhat counter-intuitive. And some functions have to be forwarded to the submanifold, like `manifold_dimension`.

Reply:
Yes, but the idea is to reuse a lot of functions. The `SymmetricMatrices` where I started trying, reduced to half in lines of code (300->150), that's what I had in mind. Maybe it's similar to a view on an array, that is also a decorator for an array showing a reduced part of it. Has to be properly documented but then it should be cool.

Reply:
That's a substantial reduction 👍 . I think I'll just have to get used to it.